go/cmd/dolt/commands/sqlserver/server.go

```diff
@@ -147,6 +149,19 @@ func Serve(
 	}
 	defer sqlEngine.Close()
 
+	// Add superuser
+	if sqlEngine.GetUnderlyingEngine().Analyzer.Catalog.MySQLDb.UserTable().Data().Count() != 0 {
+		// privileges specified, only add if superuser specified is not an existing user
+		userSpecified := config.ServerUser != defaultUser || config.ServerHost != defaultHost || config.ServerPass != defaultPass
+		superuser := sqlEngine.GetUnderlyingEngine().Analyzer.Catalog.MySQLDb.GetUser(config.ServerUser, "%", false)
+		if userSpecified && superuser == nil {
+			sqlEngine.GetUnderlyingEngine().Analyzer.Catalog.MySQLDb.AddSuperUser(config.ServerUser, "%", config.ServerPass)
+		}
+	} else {
+		// no privileges, must add superuser; will already be defaulted to root
+		sqlEngine.GetUnderlyingEngine().Analyzer.Catalog.MySQLDb.AddSuperUser(config.ServerUser, "%", config.ServerPass)
```

Comment:
Shouldn't we be using `config.ServerHost`? Otherwise it looks like it's getting completely ignored. I'm assuming the default value is `localhost`, so it should be `%` (if it's not possible to have an empty default).

Reply:
We should, but the Sysbench and Compatibility tests fail with permission denied.
Default is localhost.

Reply:
We can modify both the sysbench and compatibility tests if necessary. They should be able to work with empty defaults though, as the resulting logic should end up equivalent unless they’re specifying a root user.

Reply:
Functionally equivalent in this particular case I mean.